pkg/provider/azure_loadbalancer.go

```diff
@@ -1296,6 +1296,73 @@ func getIdleTimeout(s *v1.Service) (*int32, error) {
 	return &to32, nil
 }
 
+func getProbeIntervalInSecondsAndNumOfProbe(s *v1.Service) (*int32, *int32, error) {
+	// get number of probes
+	numberOfProbes, err := getInt32FromAnnotations(s.Annotations, consts.ServiceAnnotationLoadBalancerHealthProbeNumOfProbe, func(val *int32) error {
+		//total probe should be less than 120, minimum probe interval is 2 and minimum number of unhealthy responses is 2
+		//number of probe values should range between 2 to 60
+		if *val > 60 || *val < 2 {
+			return fmt.Errorf("total probe should be less than 120 and minimum probe interval is 2, %s value should fall into the range between 2 to 60", consts.ServiceAnnotationLoadBalancerHealthProbeNumOfProbe)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	// if numberOfProbes is not set, set it to default instead ref: https://docs.microsoft.com/en-us/rest/api/load-balancer/load-balancers/create-or-update#probe
+	if numberOfProbes == nil {
+		numberOfProbes = to.Int32Ptr(2)
+	}
+
+	probeInterval, err := getInt32FromAnnotations(s.Annotations, consts.ServiceAnnotationLoadBalancerHealthProbeInterval, func(val *int32) error {
+		//total probe should be less than 120
+		if (*val)*(*numberOfProbes) >= 120 {
+			return fmt.Errorf("total probe should be less than 120, please adjust interval and number of probe accordingly")
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	// if probeInterval is not set, set it to default instead ref: https://docs.microsoft.com/en-us/rest/api/load-balancer/load-balancers/create-or-update#probe
+	if probeInterval == nil {
+		probeInterval = to.Int32Ptr(5)
+	}
+	return probeInterval, numberOfProbes, nil
+}
+
+type Int32BusinessValidator func(*int32) error
```

Comment:
Could you add some comments for newly added types and functions?

Reply:
added